repo/blob/s3/s3_storage.go

```
@@ -182,6 +183,29 @@ func (s *s3Storage) putBlob(ctx context.Context, b blob.ID, data blob.Bytes, opt
 		StorageClass:    storageClass,
 		RetainUntilDate: retainUntilDate,
 		Mode:            retentionMode,
+		ServerSideEncryption: func() encrypt.ServerSide {
+			if s.ServerSideEncryption == "" {
+				return nil
+			}
+			switch s.ServerSideEncryption {
+			case "AES256":
+				return encrypt.NewSSE()
+			case "aws:kms":
+				var err error
+				var sse encrypt.ServerSide
+				if s.KMSKeyID != "" {
+					sse, err = encrypt.NewSSEKMS(s.KMSKeyID, nil)
+				} else {
+					sse, err = encrypt.NewSSEKMS("", nil)
+				}
+				if err != nil {
```

This is silently ignoring the error, is that intended?

---

es, you're right about the error handling. I've moved the encryption logic to a new getServerSideEncryption() function which properly handles and propagates errors